src/rdkafka.h

```diff
@@ -1727,7 +1727,69 @@ void rd_kafka_conf_set_stats_cb(rd_kafka_conf_t *conf,
 						  size_t json_len,
 						  void *opaque));
 
+/**
+ * @brief Set SASL/OAUTHBEARER token refresh callback in provided conf object.
+ *
+ * The SASL/OAUTHBEARER token refresh callback is triggered via rd_kafka_poll()
+ * whenever OAUTHBEARER is the SASL mechanism and a token needs to be retrieved,
+ * typically based on the configuration defined in \c sasl.oauthbearer.config.
+ * Function arguments:
+ *   - \p rk - Kafka handle
+ *   - \p opaque - Application-provided opaque set via rd_kafka_conf_set_opaque()
+ * 
+ * The callback must invoke \c rd_kafka_oauthbearer_set_token()
+ * or \c rd_kafka_oauthbearer_set_token_failure() to indicate success
+ * or failure, respectively.
+ * 
+ * The refresh operation is eventable and may be received via
+ * rd_kafka_queue_poll() with an event type of
+ * RD_KAFKA_EVENT_OAUTHBEARER_TOKEN_REFRESH.
+ *
+ * Note that before any broker connection is attempted the application must call
+ * rd_kafka_oauthbearer_set_token() once -- either directly or, more typically,
+ * by invoking either rd_kafka_poll() or rd_kafka_queue_poll() -- in order to
+ * cause retrieval of an initial token to occur.
+ */
+RD_EXPORT
+void rd_kafka_conf_set_oauthbearer_token_refresh_cb(rd_kafka_conf_t *conf,
+                void (*oauthbearer_token_refresh_cb) (rd_kafka_t *rk,
+                void *opaque));
 
+/**
+ * @brief Set SASL/OAUTHBEARER token and metadata
+ *
+ * The SASL/OAUTHBEARER token refresh callback or event handler must invoke
+ * this method upon success. The md_lifetime_ms value is when the token expires,
+ * in terms of the number of milliseconds since the epoch.  The extension_size
+ * should be a non-negative multiple of 2. The extension keys must not include
+ * the reserved key "auth", and all extension keys and values must conform to the
+ * required format as per https://tools.ietf.org/html/rfc7628#section-3.1:
+ * key            = 1*(ALPHA)
+ * value          = *(VCHAR / SP / HTAB / CR / LF )
+ * 
+ * @returns RD_KAFKA_RESP_ERR_NO_ERROR on success, otherwise errstr set and:
+ *          RD_KAFKA_RESP_ERR__INVALID_ARG if any of the arguments are invalid,
+ *          RD_KAFKA_RESP_ERR__NOT_IMPLEMENTED if SASL/OAUTHBEARER is disabled
```

Comment:
Reused `RD_KAFKA_RESP_ERR__STATE` so that it looks like this:

```
 * @returns RD_KAFKA_RESP_ERR_NO_ERROR on success, otherwise
 *          RD_KAFKA_RESP_ERR__NOT_IMPLEMENTED if SASL/OAUTHBEARER is not
 *              supported by this build;
 *          RD_KAFKA_RESP_ERR__STATE if SASL/OAUTHBEARER is supported but is
 *              not configured as the client's authentication mechanism.
```

As per the above comment in `rdkafka.c`, let me know if reusing `RD_KAFKA_RESP_ERR__STATE` is acceptable, and if not I will add a new code (presumably `ERR__DISABLED`, but might `ERR__NOT_CONFIGURED` or something else that directly refers to the configuration as the issue be clearer?).